### PR TITLE
✨ Source Instagram: add new fields for streams StoryInsights and MediaInsights

### PR DESCRIPTION
**Pull request by im-voracity**
## What

New metrics for StoryInsights and MediaInsights streams:
`StoryInsights`: `follows`, `profile_visits`, `shares`,  `total_interactions`.
`MediaInsights`: `follows`, `profile_visits`, `clips_replay_count`, `ig_reels_aggregated_all_plays_count`.
Also changed description for `ig_reels_avg_watch_time` and `ig_reels_video_view_total_time` that were wrong, since Instagram uses milliseconds for those metrics and not seconds.
Issue: #40610

## How

Added new fields to requests query, updated schemas and tests to reflect the changes.

## Review guide

1. source-instagram/schemas/media_insights.json
2. source-instagram/schemas/story_insights.json
3. source-instagram/source_instagram/manifest.yaml
4. source-instagram/unit_tests/resource/
5. source-instagram/unit_tests/resource/http/response/media_insights_for_carousel_album.json
6. source-instagram/unit_tests/resource/http/response/media_insights_for_general_media.json
7. source-instagram/unit_tests/resource/http/response/media_insights_for_reels.json
8. airbyte-integrations/connectors/source-instagram/unit_tests/resource/http/response/media_insights_for_video.json
9. source-instagram/unit_tests/resource/http/response/story_insights_for_story_insights_happy_path.json

## User Impact

New metrics available for affected streams

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

Context: Some metrics were deprecated in newer versions of instagram API and although the connector removed the deprecated fields, it didn't added the new metrics available. This PR focus on bringing some of them, since the remaining ones are more complex.

API Reference for all the metrics available can be found [here](https://developers.facebook.com/docs/instagram-platform/instagram-graph-api/reference/ig-media/insights/).

### CODE REVIEW

**Comment by natikgadzhi** [comment]
Thank you for this! Let me run CI on this and fix formatting, let's see how things look then

**Comment by im-voracity** [comment]
Do I have to do something to allow you to do run this?

It seems it tried to run on my repo and got an error because of that

File: airbyte-integrations/connectors/source-instagram/pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ requires = [ "poetry-core>=1.0.0",]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
-version = "3.0.20"
+version = "3.1.0"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-instagram/pyproject.toml:6]: It is missing to bump the version in the metadata.yaml file too. Can you do that?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-instagram/pyproject.toml (hunk 1 above)
Comment (im-voracity) [reply airbyte-integrations/connectors/source-instagram/pyproject.toml:6]: Done

**Comment by im-voracity** [comment]
Any updates on this review?

**Comment by natikgadzhi** [comment]
The changes make sense to me, but the team that owns Instagram is pretty backlogged. ;( 

@bazarnov maybe I can convince you resolve the conflicts and take a look? ;)

**Comment by bazarnov** [comment]
@natikgadzhi No problem; I'll take a look. However, this is a private repo I don't have access to. Thus, the conflicts must be resolved by the repo maintainer @im-voracity (I believe this is you). Merging with master (pull) is enough to list and resolve the conflicts (there are a couple of lines of code to accept and move your changes on top of it).

**Review by bazarnov — APPROVED** [review]
The changes look good to me, this is not a Breaking Change, thus we should be safe to proceed. 

@im-voracity Poke @marcosmarxm when the conflicts are resolved from your side, please. 

Thank you for the contribution!

**Review by marcosmarxm — APPROVED** [review]
LGTM!

**Comment by marcosmarxm** [comment]
Hello your feedback matters a lot to us. Can you spare just 2 minutes to complete a [survey](https://form.typeform.com/to/DOBqejte)? We’re dedicated to making the contributor experience even better, and your input is key to achieving excellence. Thank you for helping us improve!

**Pull request merged**